tests/test_dia_object.py

```diff
@@ -20,28 +20,141 @@
 # see <http://www.lsstcorp.org/LegalNotices/>.
 #
 
+import numpy as np
 import unittest
+
+from lsst.ap.association import *
+import lsst.afw.table as afwTable
+import lsst.afw.geom as afwGeom
 import lsst.utils.tests
 
 
-class TestDIAObject(unittest.TestCase):
+def create_test_dia_sources(n_sources=5):
+    """ Create dummy DIASources for use in our tests.
 
-    def setUp(self):
-        pass
+    Parameters
+    ----------
+    n_sources : int (optional)
+        Number of fake sources to create for testing.
 
-    def tearDown(self):
-        pass
+    Returns
+    -------
+    A lsst.afw.SourceCatalog
+    """
+    sources = afwTable.SourceCatalog(afwTable.SourceTable.makeMinimalSchema())
 
-    def test_initialization(self):
-        pass
+    for src_idx in range(n_sources):
+        src = sources.addNew()
+        src['id'] = src_idx
+        src['coord_ra'] = afwGeom.Angle(0.0 + 1. * src_idx,
+                                        units=afwGeom.degrees)
+        src['coord_dec'] = afwGeom.Angle(0.0 + 1. * src_idx,
+                                         units=afwGeom.degrees)
+        # Add a flux at some point
 
-    def test_dia_source_append(self):
-        pass
+    return sources
 
-    def compute_light_curve(self):
-        pass
 
-    def compute_summary_statistics(self):
+class TestDIAObject(unittest.TestCase):
+
+    def setUp(self):
+        """ Create the DIAObjct schema we will use throughout these tests.
+        """
+        self.dia_obj_schema = make_minimal_dia_object_schema()
+
+    def tearDown(self):
+        """ Delete the schema when we are done.
+        """
+        del self.dia_obj_schema
+
+    def test_init(self):
+        """ Test DIAObject creation and if we can properly instantiate a
+        DIAObject from an already create dia_object_record.
+        """
+        single_source = create_test_dia_sources(1)
+
+        dia_obj = DIAObject(single_source, None)
+        dia_obj_dup = DIAObject(single_source, dia_obj.dia_object_record)
+
+        self._compare_dia_object_values(dia_obj, 0, 0.0, np.nan)
+        self._compare_dia_object_values(dia_obj_dup, dia_obj.id, 0.0, np.nan)
+
+    def _compare_dia_object_values(self, dia_object, expected_id,
+                                   expected_mean, expected_std):
+        """ Compare values computed in the compute_summary_statistics
+        DIAObject class method with those expected.
+
+        Parameters
+        ----------
+        dia_object : lsst.ap.association.DIAObject
+            Input DIAObect to test
+        expected_id : int
+            Expected id of the DIAObject
+        expected_mean : float
+            Expected mean value of the parameters to be tested.
+        expected_std : float
+            Expected standard diviation of the DIAObject.
+
+        Return
+        ------
+        None
+        """
+
+        for name in self.dia_obj_schema.getNames():
```

Comment:
This really doesn't need to be a loop; you already know the names of the things you want to compare to.